pkg/ccl/backupccl/alter_backup_schedule.go

```diff
@@ -0,0 +1,401 @@
+	}
+	recurrenceStr, err := recurrence()
+	if err != nil {
+		return nil, nil, err
+	}
+	if incJob != nil {
+		if err := incJob.SetSchedule(recurrenceStr); err != nil {
+			return nil, nil, err
+		}
+	} else {
+		if err := fullJob.SetSchedule(recurrenceStr); err != nil {
+			return nil, nil, err
+		}
+	}
+	return fullJob, incJob, nil
+}
+
+func processFullBackupRecurrence(
+	ctx context.Context,
+	p sql.PlanHookState,
+	fullBackupAlways bool,
+	fullBackupRecurrence func() (string, error),
+	isEnterpriseUser bool,
+	fullJob *jobs.ScheduledJob,
+	fullArgs *backuppb.ScheduledBackupExecutionArgs,
+	incJob *jobs.ScheduledJob,
+	incArgs *backuppb.ScheduledBackupExecutionArgs,
+) (
+	*jobs.ScheduledJob,
+	*backuppb.ScheduledBackupExecutionArgs,
+	*jobs.ScheduledJob,
+	*backuppb.ScheduledBackupExecutionArgs,
+	error,
+) {
+	var err error
+
+	if !fullBackupAlways && fullBackupRecurrence == nil {
+		return fullJob, fullArgs, incJob, incArgs, nil
+	}
+
+	env := sql.JobSchedulerEnv(p.ExecCfg())
+	ex := p.ExecCfg().InternalExecutor
+	if fullBackupAlways {
+		if incJob == nil {
+			// Nothing to do.
+			return fullJob, fullArgs, incJob, incArgs, nil
+		}
+		// Copy the cadence from the incremental to the full, and delete the
+		// incremental.
+		if err := fullJob.SetSchedule(incJob.ScheduleExpr()); err != nil {
+			return nil, nil, nil, nil, err
+		}
+		fullArgs.DependentScheduleID = 0
+		fullArgs.UnpauseOnSuccess = 0
+		if err := incJob.Delete(ctx, ex, p.Txn()); err != nil {
+			return nil, nil, nil, nil, err
+		}
+		incJob = nil
+		incArgs = nil
+		return fullJob, fullArgs, incJob, incArgs, nil
+	}
+
+	// We have FULL BACKUP <cron>.
+	if !isEnterpriseUser {
+		return nil, nil, nil, nil, errors.Newf("Enterprise license required to use incremental backups. " +
+			"To modify the cadence of a full backup, use the 'RECURRING <cron>' clause instead.")
+	}
+
+	if incJob == nil {
+		// No existing incremental job, so we need to create it, copying details
+		// from the full.
+		node, err := parser.ParseOne(fullArgs.BackupStatement)
+		if err != nil {
+			return nil, nil, nil, nil, err
+		}
+		stmt, ok := node.AST.(*tree.Backup)
+		if !ok {
+			return nil, nil, nil, nil, errors.Newf("unexpected node type %T", node)
+		}
+		stmt.AppendToLatest = true
+
+		scheduleExprFn := func() (string, error) {
+			return fullJob.ScheduleExpr(), nil
+		}
+		incRecurrence, err := computeScheduleRecurrence(env.Now(), scheduleExprFn)
+		if err != nil {
+			return nil, nil, nil, nil, err
+		}
+
+		incJob, incArgs, err = makeBackupSchedule(
+			env,
+			p.User(),
+			fullJob.ScheduleLabel(),
+			incRecurrence,
+			*fullJob.ScheduleDetails(),
+			jobs.InvalidScheduleID,
+			fullArgs.UpdatesLastBackupMetric,
+			stmt,
+			fullArgs.ChainProtectedTimestampRecords,
+		)
+
+		if err != nil {
+			return nil, nil, nil, nil, err
+		}
+
+		// We don't know if a full backup has completed yet, so pause incremental
+		// until a full backup completes.
+		incJob.Pause()
```

Comment:
Huh, I'm not sure we want to pause it here? like, say we do 10min inc and we were doing daily full but need to switch to 6h full since the chains are getting long; we do alter ... full backup */6 * ... etc, and then I think we expect our next full to run at 1800 instead of 0000, but we probably expect our inc backups to carry on on their 10min cadence as if nothing changed, no?

Reply:
We can only get to this branch if incJob == nil, that is, we were doing _no_ incremental backups of any kind before, but we're adding some now. I think `PAUSE` is right in that case, no? We could do fancy logic to see if the FULL schedule has been run before, I suppose, but this does seem simpler and quite reasonable.